app/notifications/validators.py

```diff
@@ -33,6 +34,7 @@ def check_service_over_api_rate_limit(service, api_key):
             raise RateLimitError(rate_limit, interval, api_key.key_type)
 
 
+@statsd_catch(namespace="validators", counter_name="rate.limit.daily", exception=TooManyRequestsError)
```

Comment:
I'm not convinced by the name. I fee like introducing 2 namespaces `rate.limit` is too much, "rate limit" should be a thing on its own. Maybe `rate_limit.service_daily`? Do we have other counters in the codebase related to rate limit?

Reply:
Other counters in *api* project so far:

```
'callback.{}.{}'
'callback.sendgrid.{}'
redis.daily_limit_cache_key(service.id)
'callback.sns.{notification_status}'
'callback.ses.{}'
'callback.ses-smtp.{}'
"user-agent.{}"
"clients.ses.error"
"clients.ses.error"
"clients.ses.error"
"clients.ses.success"
"clients.sendgrid.error"
"clients.sendgrid.success"
"clients.sns.error"
"clients.sns.error"
"clients.sns.success"
"clients.sns.error"
"clients.pinpoint.error"
"clients.pinpoint.error"
"clients.pinpoint.error"
"clients.pinpoint.success"
"clients.pinpoint.error"
```

Reply:
Ah naming, one of these hard things.. I will make that change. 

Just one comment to make sure we agree on metrics naming and goals around it.

I want to represent hierarchies of rates using many metrics that we'll be able to catch with this annotation. For example, in the same *validators* module, we could have right now `rate.limit.api` for the `check_service_over_api_rate_limit` method.  

We should be careful in how we name these because we can perform aggregate operations on hierarchies of metrics (which the parent-child associations are represented with `.`s), so that all families of business related metrics should be grouped together.

That said, I am not sure the name I got is the right one... most probably not and we'll not get this right the first time. I would say let's review all these names when we start leveraging statsd and metrics dashboard in a robust manner.